### DISCUSSION TITLE
Improvements for Jenkins integration documentation

### DISCUSSION THREAD

**Discussion by d3rp3tt3** [dagger/dagger#12511] [Ideas]
### What is the issue?

This issue will collect the various improvements we can make in the near term to the documentation for using Jenkins and Dagger together for CI. 

### Summary of issues
* The Node and Go SDKs have examples of how to use CI with Dagger, but the CLI and Python SDKs do not
* Users must know how to specify where the Dagger CLI gets installed. The docs do not detail that Dagger's uses the $XDG_CACHE_HOME env variable to find/download the CLI. If that variable is not set, it falls back to $HOME to install the CLI.

### Improvements
- [X] ~Add CI examples for the Python SDK~ We have here https://docs.dagger.io/454108/python-ci
- [ ] Add CI examples for the CLI
- [ ] Update Jenkins examples to detail what each command in the bash script is doing
- [ ] Add Jenkins instructions for specifying an environment variable for Dagger's CLI download location

### Resources
* Discord link: https://discord.com/channels/707636530424053791/1082381372460646481

### Call to action to the Dagger community
If there are things you'd like to include or clarify in the Jenkins instructions, please don't hesitate to comment on this issue, and I'll add them to the description to track.

**Comment by d3rp3tt3** [comment]
Example: setting the environment variable to specify where Dagger should be installed: `export XDG_CACHE_HOME=${env.YOUR_LOCATION};`

**Jenkins declarative pipeline**

```
pipeline {
    agent any

    stages {
        stage('build image') {
            steps {
                git branch: 'master', url: 'https://github.com/robertu94/libpressio'

                sh '''#!/bin/bash
                python3.10 -m venv workenv;
                . ./workenv/bin/activate;
                pip3.10 install "dagger-io";
                export XDG_CACHE_HOME=${env.WORKSPACE};
                python3.10 ./ci/test.py
                '''     
            }
        }
    }
}
```

**Comment by robertu94** [comment]
Another thing that was missing in my case was on CentOS 8, there were also kernel modules `iptable_nat` and `iptable_filter` that needed to be loaded.
